### ISSUE TITLE
Add transparency to builds with env "CI"

### ISSUE THREAD

**Issue by Undestor** [ng-packagr/ng-packagr#2730]
## Type of Issue

```
[ ] Bug Report
[X] Feature Request
```

## Description

In one of my projects we stumbled upon a strange behaviour where our ci pipeline produces a different build result, than our local environments. In our local environments the angular application build failed because of an error:
`NG3001: Unsupported private class SomeComponent. This class is visible to consumers via SomeModule -> SomeComponent, but is not exported from the top-level library entrypoint` (https://github.com/angular/angular/blob/c62b2dae8c4653996969df992e8f88887e7c83c4/packages/compiler-cli/src/ngtsc/entry_point/src/private_export_checker.ts#L38C7-L38C7)
But our ci pipeline produced no error at all (our desired outcome was a failure in the ci build)

After multiple days of digging we finally came to the conclusion that the build does different steps depending on the environment variable "CI" and "GITLAB_CI" respectively.
During the CI build the angular diagnostics seemed to be ignored:
https://github.com/ng-packagr/ng-packagr/blob/91f4605e80564af307ced9edf76808c7d44a66ad/src/lib/ngc/compile-source-files.ts#L163

Which lead us to the questions:
- How should we have known that the builds behaves different on a CI environment? Did me miss some documentation or parameters?
- How can we achieve the same output for the local environment and the ci environment without changing environment variables? In my opinion there have to be some config variables to navigate the different behaviours. Reacting to environment variables is non transparent for the user.

I know i sneaked some questions in the feature request (add transparency on how the build behaves on the CI environment), sorry for that :)

#### How To Reproduce

- Force a diagnostic error like NG3001
- Build the application without the environment variables "CI", "GITLAB_CI", ... => Build fails
- Build the application with the environment variable CI=true => Build succeeds

#### Expected Behaviour

- Documentation on how and why the build differs
- Documentation (and implementation?) on how the different behaviours can be navigated

#### Version Information

```
$ node_modules/.bin/ng-packagr --version
ng-packagr: 16.2.2
@angular/compiler: 16.2.3
rollup: 3.26.0
typescript: 5.1.6
```

**Comment by JoostK** [comment]
This sounds like a bug to me

**Comment by alan-agius4** [comment]
This has been solved in version 17.0.1
